src/control/lib/control/pool.go

```diff
@@ -1057,89 +1063,212 @@ func processSCMSpaceStats(log debugLogger, filterRank filterRankFn, scmNamespace
 			scmBytes = usableBytes
 		}
 
-		if _, exists := rankNVMeFreeSpace[scmNamespace.Mount.Rank]; exists {
-			return 0, errors.Errorf("Multiple SCM devices found for rank %d",
+		if _, exists := req.rankNVMeFreeSpace[scmNamespace.Mount.Rank]; exists {
+			return 0, errors.Errorf("multiple scm devices found for rank %d",
 				scmNamespace.Mount.Rank)
 		}
 
 		// Initialize entry for rank in NVMe free space map.
-		rankNVMeFreeSpace[scmNamespace.Mount.Rank] = 0
+		req.rankNVMeFreeSpace[scmNamespace.Mount.Rank] = 0
```

Comment:
Sort of weird to me that a function focused on SCM initializes the NVMe free space map.

Reply:
agree, it's not something I've added in this change but I will refactor to separate concerns